bin/parallel.py

```diff
@@ -163,6 +163,15 @@ def _worker(self, cores: bool | list[int] = False):
     def _interrupt_handler(self, sig, frame):
         util.fatal('Running interrupted', force=True)
 
+    def _handle_first_error(self):
+        if self.first_error is not None:
+            first_error = self.first_error
+            self.first_error = None
+            # we are the main thread now, so we can handle this
+            if isinstance(first_error, ChildProcessError):
```

Comment:
can you expand the comment? I don't really understand this if statement and why we must forward to the original interrupt handler here. Why are we the main thread?
In which circumstances does the `if` trigger?

Reply:
in `util.py:exec_command` we throw a `ChildProcessError` if the child is terminated by `ctrl+c` and we are not the main process (If we are the main process we call fatal... i don't know why this distinction is made...).
Anyway, this means a `ChildProcessError` is only thrown if we use the parallel queue.
And the thread that calls `join()` or `done()` is supposed to handle this error, but there is nothing to handle we should exit since someone pressed `ctrl+c`.

Reply:
I suppose the distinction is so that the main thread can clearly handle errors from child processes somehow.